feat(docs): update check results page with retries [sc-14874] (#791)
checkly/checklyhq.com: site/content/docs/monitoring/check-results.md

### Summary section

> The summary at the top of the page allows for filtering based on all of the page's data points and the selected timeframe and locations. Retried check runs do not influence this section and only final results are taken into account.

Reviewer: Suggestion for conciseness. 
```suggestion
The summary at the top of the page allows for filtering based on the page's data points and the selected timeframe and locations. Retried check runs do not influence this section; only the final results are considered.
```

Selecting a check execution will take you to the dedicated check result page for it, which will look different based on whether you had been running a [Browser](#browser-check-results) or an [API check](#api-check-results).

When retries are enabled, an additional icon highlights that the check result contains multiple check runs. All attempts as well as the final result are available to view.

> The check retries ratio indicates how many of the total checks run were executed twice due to an initial failed attempt.

Reviewer: I'm not sure I like this suggestion more TBH - it's just a way to get around having to update this again if we add the capability of having more than 1 retry per check run.
```suggestion
The check retries ratio indicates how many of the total checks run were retried due to an initial failed attempt.
```

    d. First Byte

    e. Download

## Check results with retries

> When retries are enabled through the "Double-check on failure" setting, tabs will indicate that the check result contains multiple check runs:

Reviewer: ```suggestion
When retries are enabled through the "[Double-check on failure](/docs/alerting/#double-checking)" setting, tabs will indicate that the check result contains multiple check runs:
```

1. The initial failed attempt
2. The final result (which may have failed or succeeded)

> By selecting a check run all data and assets are available for inspection for each attempt.

Reviewer 2: ```suggestion
When selecting a check run, all data and assets are available for inspection for each attempt.
```
